spring-kafka/src/main/java/org/springframework/kafka/requestreply/ReplyingKafkaTemplate.java

```diff
@@ -240,8 +287,16 @@ public void onMessage(List<ConsumerRecord<K, R>> data) {
 			else {
 				RequestReplyFuture<K, V, R> future = this.futures.remove(correlationId);
 				if (future == null) {
-					this.logger.error("No pending reply: " + record + " with correlationId: "
-							+ correlationId + ", perhaps timed out");
+					if (this.sharedReplyTopic) {
+						if (this.logger.isDebugEnabled()) {
+							this.logger.debug("No pending reply: " + record + " with correlationId: "
+									+ correlationId + ", perhaps timed out, or using a shared reply topic");
+						}
+					}
+					else if (this.logger.isErrorEnabled()) {
+						this.logger.error("No pending reply: " + record + " with correlationId: "
+								+ correlationId + ", perhaps timed out, or using a shared reply topic");
```

Comment:
Looks like message is the same for both logging.
Can't we extract it into the `private` method?